### PR TITLE
[MI-180] Add User-Agent header to API calls

### PR DESCRIPTION
**Pull request by joseconsador**
Add User-Agent header to API calls
Describe your PR

/cc @miogalang @jwswj @mmolina @atroche

### References
 - Jira link: https://zendesk.atlassian.net/browse/MI-180

### Risks
 - Low - the User-Agent header might not be sent to the API

### CODE REVIEW

File: tests/Zendesk/API/UnitTests/Core/ResourceTest.php [hunk 1]
```diff
@@ -255,4 +255,23 @@ public function testHandlesApiException()
...
+    {
+        $this->assertEndpointCalled(
+            function () {
+                $this->dummyResource->findAll();
+            },
+            'dummy_resource.json',
+            'GET',
+            [
+                'headers' => [
+                    'User-Agent' => "ZendeskAPI PHP {$this->client->getClientVersion()}"
```
Comment (jwswj) [inline tests/Zendesk/API/UnitTests/Core/ResourceTest.php]: Could be good to mock `$this->client->getClientVersion()` so you're tested the exact string being returned.

File: src/Zendesk/API/HttpClient.php [hunk 2]
```diff
@@ -257,6 +261,27 @@ public function setAuth($strategy, array $options)
...
+     */
+    public function getClientVersion()
+    {
+        return $this->clientVersion;
+    }
+
+    /**
+     * @param string $clientVersion
+     */
+    public function setClientVersion($clientVersion)
```
Comment (jwswj) [inline src/Zendesk/API/HttpClient.php]: This doesn't look like it's being used?

File: src/Zendesk/API/HttpClient.php [hunk 3]
```diff
@@ -94,6 +94,11 @@ class HttpClient
     use InstantiatorTrait;
 
     /**
+     * @var string
+     */
+    private $clientVersion = '2.0.0';
```
Comment (jwswj) [inline src/Zendesk/API/HttpClient.php]: Could this be a constant?

Comment (miogalang) [inline src/Zendesk/API/HttpClient.php]: Agree. This should be a constant

**Comment by joseconsador** [comment]
:haircut: 
@miogalang @jwswj

File: tests/Zendesk/API/UnitTests/Core/ResourceTest.php [hunk 4]
```diff
@@ -255,4 +256,38 @@ public function testHandlesApiException()
...
+
+        $this->assertEndpointCalled(
+            function () use ($dummyResource, $mockUserAgent) {
+                $dummyResource->findAll();
+            },
+            'dummy_resource.json',
+            'GET',
+            [
+                'headers' => [
+                    'User-Agent' => $mockUserAgent
```
Comment (jwswj) [inline tests/Zendesk/API/UnitTests/Core/ResourceTest.php]: Looking at this again it feels like overkill compared to https://github.com/zendesk/zendesk_api_client_rb/blob/master/spec/core/configuration_spec.rb#L17

Comment (joseconsador) [inline tests/Zendesk/API/UnitTests/Core/ResourceTest.php]: :haircut: applied something similar

**Comment by jwswj** [comment]
:+1:

**Comment by miogalang** [comment]
:+1:

**Pull request merged**